error[E0080]: constant evaluation error
  --> $DIR/issue-44578.rs:24:24
   |
LL |     const AMT: usize = [A::AMT][(A::AMT > B::AMT) as usize]; //~ ERROR E0080
   |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ index out of bounds: the len is 1 but the index is 1

error[E0080]: constant evaluation error
  --> $DIR/issue-44578.rs:14:5
   |
LL |     const AMT: usize;
   |     ^^^^^^^^^^^^^^^^^
...
LL |     const AMT: usize = [A::AMT][(A::AMT > B::AMT) as usize]; //~ ERROR E0080
   |                        ------------------------------------ index out of bounds: the len is 1 but the index is 1

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0080`.
